src/ir/module-utils.cpp

```diff
@@ -54,17 +54,15 @@ struct CodeScanner
     if (auto* call = curr->dynCast<CallIndirect>()) {
       counts.note(call->heapType);
     } else if (auto* call = curr->dynCast<CallRef>()) {
-      if (call->isReturn && call->target->type.isFunction()) {
```

Comment:
The `isReturn` change makes us count in the way that is correct in the spec but we haven't yet moved to, I think? But the idea is that there's no harm to doing that for now, and eventually this code will become right in binaryen as well?

Reply:
We emit a type annotation for non-return `call_ref` as well since #5079, so this is necessary for correctness.